python/tank/platform/software_launcher.py

```diff
@@ -336,47 +375,83 @@ def get_standard_plugin_environment(self):
 
         return env
 
-    def is_version_supported(self, version):
+    def is_version_supported(self, sw_version):
         """
-        Compares the given version string with the :meth:`~minimum_supported_version`
-        and returns a boolean to indicate whether it is supported by the launcher
-        or not.
+        Inspects the supplied :class:`SoftwareVersion` object to see if aligns
+        with this launcher's known product and version limitation. Will check
+        the :meth:`~minimum_supported_version` as well as the list of product
+        and version filters.
+
+        :param sw_version: :class:`SoftwareVersion` object to test against the
+            launcher's product and version limitations.
 
-        :param str version: Version string to test, e.g. "2015" or "2017.3.sp4"
         :returns: True if supported, False otherwise.
         """
-        if self.minimum_supported_version is None:
-            return True
 
-        if is_version_older(version, self.minimum_supported_version):
+        # check the minimum version
+        min_version = self.minimum_supported_version
+        if min_version and is_version_older(sw_version.version, min_version):
+            self.logger.debug(
+                "Executable %s not supported. Failed minimum check. (%s < %s)" %
+                (sw_version.path, sw_version.version, min_version)
+            )
             return False
-        else:
-            return True
 
+        # check versions list
+        if self.versions and sw_version.version not in self.versions:
+            self.logger.debug(
+                "Executable %s not supported. Failed version check. (%s not in %s)" %
+                (sw_version.path, sw_version.version, self.versions)
+            )
+            return False
+
+        # check products list
+        if self.products and sw_version.product not in self.products:
+            self.logger.debug(
+                "Executable %s not supported. Failed product check. (%s not in %s)" %
+                (sw_version.path, sw_version.product, self.products)
+            )
+            return False
 
+        # passed all checks. must be supported!
+        return True
 
 class SoftwareVersion(object):
```

Comment:
How about ProductInstance?

It's about a very specific instance of a product.

For example:
- Nuke X
- 10v0.1
- Non-commercial